python/ray/tests/conftest.py

```diff
@@ -88,12 +82,49 @@ def external_redis(request, monkeypatch):
         processes.append(proc)
         wait_for_redis_to_start("127.0.0.1", port, ray_constants.REDIS_DEFAULT_PASSWORD)
     address_str = ",".join(map(lambda x: f"127.0.0.1:{x}", external_redis_ports))
-    monkeypatch.setenv("RAY_REDIS_ADDRESS", address_str)
-    yield None
+    yield address_str
     for proc in processes:
         proc.process.terminate()
 
 
+@pytest.fixture
+def maybe_external_redis(request):
+    import os
+
+    if "REDIS_MODE" in os.environ:
+        with _setup_redis(request) as addr:
```

Comment:
Can this reuse `external_redis()`?

Reply:
The problem here is that for some tests, we need to set up Redis. Any suggestions?

Reply:
Would `yield from external_redis()` work?